test/e2e/framework/addon/vault/setup.go

```diff
@@ -343,6 +397,88 @@ func (v *VaultInitializer) setupRole() error {
 	return nil
 }
 
+func (v *VaultInitializer) setupKubernetes() error {
+	if len(v.APIServerURL) == 0 {
+		// skip initialization if not provided
+		return nil
+	}
+
+	// vault auth-enable kubernetes
+	auths, err := v.client.Sys().ListAuth()
+	if err != nil {
+		return fmt.Errorf("Error fetching auth mounts: %s", err.Error())
+	}
+
+	if _, ok := auths[v.KubernetesAuthPath+"/"]; !ok {
+		options := &vault.EnableAuthOptions{Type: "kubernetes"}
+		if err := v.client.Sys().EnableAuthWithOptions(v.KubernetesAuthPath, options); err != nil {
+			return fmt.Errorf("Error enabling kubernetes auth: %s", err.Error())
+		}
+	}
+
+	// vault write auth/kubernetes/config
+	params := map[string]string{
+		"kubernetes_host":    v.APIServerURL,
+		"kubernetes_ca_cert": v.APIServerCA,
+	}
+
+	url := fmt.Sprintf("/v1/auth/%s/config", v.KubernetesAuthPath)
+	_, err = v.callVault("POST", url, "", params)
+
+	if err != nil {
+		return fmt.Errorf("error configuring kubernetes auth backend: %s", err.Error())
+	}
+
+	return nil
+}
+
+// CreateKubernetesRole creates a service account and ClusterRoleBinding for Kubernetes auth delegation
+func (v *VaultInitializer) CreateKubernetesRole(client kubernetes.Interface, namespace, roleName, serviceAccountName string) error {
+	serviceAccount := NewVaultServiceAccount(serviceAccountName)
+	_, err := client.CoreV1().ServiceAccounts(namespace).Create(serviceAccount)
+
+	if err != nil {
+		return fmt.Errorf("error creating ServiceAccount for Kubernetes auth: %s", err.Error())
+	}
+
+	roleBinding := NewVaultServiceAccountRoleBinding(namespace, serviceAccountName)
+	_, err = client.RbacV1().ClusterRoleBindings().Create(roleBinding)
```

Comment:
Does this need to be a ClusterRoleBinding? Cluster-scoped resources are a little trickier wrt clean up, as any duplicates will cause cascading failures throughout the test suite

Reply:
If it *does*, we need to make sure each resource is uniquely named (and cleaned up! 😄)